browser/android/brave_cosmetic_resources_tab_helper.cc

```diff
@@ -0,0 +1,125 @@
+
+
+namespace {
+bool IsAdBlockEnabled(content::WebContents* contents, const GURL& url) {
+  Profile* profile = static_cast<Profile*>(contents->GetBrowserContext());
+
+  auto* map = HostContentSettingsMapFactory::GetForProfile(profile);
+  brave_shields::ControlType control_type =
+      brave_shields::GetAdControlType(map, contents->GetLastCommittedURL());
+  if (brave_shields::GetBraveShieldsEnabled(map, url) &&
+      control_type != brave_shields::ALLOW) {
+    content::RenderFrameHost::AllowInjectingJavaScript();
+
+    return true;
+  }
+
+  return false;
+}
+
+std::unique_ptr<base::ListValue> GetUrlCosmeticResourcesOnTaskRunner(
+		const std::string& url) {
+  auto result_list = std::make_unique<base::ListValue>();
+
+  base::Optional<base::Value> resources = g_brave_browser_process->
+      ad_block_service()->UrlCosmeticResources(url);
+
+  if (!resources || !resources->is_dict()) {
+    return result_list;
+  }
+
+  base::Optional<base::Value> regional_resources = g_brave_browser_process->
+      ad_block_regional_service_manager()->UrlCosmeticResources(url);
+
+  if (regional_resources && regional_resources->is_dict()) {
+    ::brave_shields::MergeResourcesInto(
+        std::move(*regional_resources), &*resources, /*force_hide=*/false);
+  }
+
+  base::Optional<base::Value> custom_resources = g_brave_browser_process->
+      ad_block_custom_filters_service()->UrlCosmeticResources(url);
+
+  if (custom_resources && custom_resources->is_dict()) {
+    ::brave_shields::MergeResourcesInto(
+        std::move(*custom_resources), &*resources, /*force_hide=*/true);
+  }
+
+  result_list->Append(std::move(*resources));
+
+  return result_list;
+}
+
+void GetUrlCosmeticResourcesOnUI(content::RenderFrameHost* render_frame_host,
+	std::unique_ptr<base::ListValue> resources) {
+  if (!resources) {
+    return;
+  }
+  for (auto i = resources->GetList().begin();
+  	  i < resources->GetList().end(); i++) {
+  	const base::DictionaryValue* resources_dict;
+	if (!i->GetAsDictionary(&resources_dict)) {
+	  continue;
+	}
+	std::string to_inject;
+	resources_dict->GetString("injected_script", &to_inject);
+	render_frame_host->ExecuteJavaScript(
+       base::UTF8ToUTF16(to_inject),
+       base::NullCallback());
+  }
+}
+}  // namespace
+
+
+BraveCosmeticResourcesTabHelper::BraveCosmeticResourcesTabHelper(
+    content::WebContents* contents)
+    : WebContentsObserver(contents) {
+}
+
+BraveCosmeticResourcesTabHelper::~BraveCosmeticResourcesTabHelper() {
+}
+
+void BraveCosmeticResourcesTabHelper::ProcessURL(
+	content::WebContents* contents, const GURL& url) {
+  if (!IsAdBlockEnabled(contents, url)) {
+    return;
+  }
+  g_brave_browser_process->ad_block_service()->GetTaskRunner()->
+  	  PostTaskAndReplyWithResult(FROM_HERE,
+          base::BindOnce(&GetUrlCosmeticResourcesOnTaskRunner, url.spec()),
+          base::BindOnce(&GetUrlCosmeticResourcesOnUI,
+                  web_contents()->GetMainFrame()));
```

Comment:
Does this work on iframes?

Reply:
Good catch, I should pass a correct frame here, but it worked like that though, on that embedded video that were a problem without iframes injections I haven't seen any ad https://www.pcmag.com/news/the-best-vr-games-for-2019. But redoing it to how it was.

Reply:
I passed a correct frame there.